doc/vlan_stacking/vlan_stacking_HLD.md

```diff
@@ -0,0 +1,642 @@
+ ### Rev 0.1
+
+# Table of Contents
+
+- [List of Tables](#list-of-tables)
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitionsabbreviations)
+- [Overview](#overview)
+  - [VLAN Stacking Deployment Use Cases](#vlan-stacking-deployment-use-cases)
+    - [QinQ Use Case](#qinq-use-case)
+    - [VLAN Translation Use Case](#vlan-translation-use-case)
+- [Architecture Design](#architecture-design)
+- [Requirements](#requirements)
+  - [Configuration](#configuration)
+- [High-Level Design](#high-level-design)
+  - [VlanMgr](#vlanmgr)
+  - [DB](#db)
+    - [CONFIG_DB](#config_db)
+    - [APPL_DB](#appl_db)
+  - [SAI API](#sai-api)
+- [Configuration and management](#configuration-and-management)
+  - [CLI/YANG model Enhancements](#cliyang-model-enhancements)
+    - [CLI](#cli)
+    - [Yang model](#yang-model)
+  - [Examples](#examples)
+    - [Examples of QinQ Configuration](#examples-of-qinq-configuration)
+    - [Examples of VLAN Translation Configuration](#examples-of-vlan-translation-configuration)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictionslimitations)
+- [Testing Requirements/Design](#testing-requirementsdesign)
+  - [System Test Cases](#system-test-cases)
+- [Future Work](#future-work)
+
+# List of Tables
+
+* [Table 1: Revision](#revision)
+* [Table 2: Definitions/Abbreviations](#definitionsabbreviations)
+* [Table 3: SAI attributes related to VLAN stacking](#sai-api)
+
+# Revision
+
+###### Table 1: Revision
+
+| Rev   |   Date     | Author        | Change Description |
+|:-----:|:----------:|:-------------:|--------------------|
+| 0.1   | 8-Oct-2020 | Tommy Tseng   | Initial version    |
+
+# Scope
+
+This document describes the high level design of VLAN stacking feature.
+
+# Definitions/Abbreviations
+
+###### Table 2: Definitions/Abbreviations
+
+| Abbreviation | Full form              |
+|--------------|------------------------|
+| VID          | Virtual LAN Identifier |
+| S-VLAN       | Service Provider VLAN  |
+| C-VLAN       | Customer VLAN          |
+| S-TAG        | S-VLAN tag             |
+| C-TAG        | C-VLAN tag             |
+
+# Overview
+
+This document describes the design details of the VLAN stacking feature. VLAN stacking is a feature designed for service providers who carry traffic of multiple customers across  their networks and are required to maintain the VLAN and Layer 2 protocol configurations of each customer without impacting the traffic of other customers. VLAN stacking include QinQ feature and VLAN Translation feature.
```

Comment:
It support. The below example show how to configure many-to-one as matching C-VLAN 10, 20 to S-VLAN 300.

{"VLAN_STACKING": {"Ethernet52|INGRESS|10": {"action": "PUSH","s_vlanid": "30"},"Ethernet52|INGRESS|20": {"action": "PUSH","s_vlanid": "30"},"Ethernet52|INGRESS|40": {"action": "PUSH","s_vlanid": "30"},"Ethernet52|EGRESS|30": {"action": "POP"}}}

Reply:
Does it mean that each and every C-VLANs must be defined explicitly? Possibly hundreds of them... How about all-to-one mode?

Reply:
Yes, it need to define the C-VLANs. Regarding to all-to-one. It's hardware dependent. If the entry has enough vlan ids, it can do the configuration. For the future work, it can change new schema to support range format for VLAN ids.

Reply:
@akokhan Is there any other comment there, ifnot, please help to approve to resolve this